straxen/plugins/peaklet_processing.py

```diff
@@ -207,6 +225,141 @@ def clip_peaklet_times(peaklets, start, end):
                 p['length'] = (end - p['time']) // p['dt']
 
 
+@numba.jit(nopython=True, nogil=True, cache=True)
+def peak_saturation_correction(records, peaks, to_pe,
```

Comment:
I'd say this would be better in strax right, it's quite a general thing

Reply:
I am in favor of this being part of straxen. As this seems to be a feature, specific to XENON detectors. It's hard to say this method can be universally applied to other non-XENON ones.

Reply:
I'm not so sure, most experiments will have ADCs. E.g. a small scale TPC like XAMS or XEBRA will both potentially also see ADC-saturation as the hardware is similar to the nT experiment.